### PR TITLE
add an interesting test

### PR DESCRIPTION
**Pull request by konnov**
This is an interesting example that appeared in the [TLA+ user group](https://groups.google.com/g/tlaplus/c/cD1bhMcHxqA).  Adding it as an integration test.

### CODE REVIEW

**Review by thpani — APPROVED** [review]
Cool, I saw this in the tlaplus group and then immediately after your PR 😄 

LGTM

**Review by Kukovec — APPROVED** [review]

**Comment by shonfeder** [comment]
Very nice! Should the TLA spec or the integration test perhaps have a link to the google group discussion so we can trace the context?

File: test/tla/TestHash2.tla [hunk 1]
```diff
@@ -0,0 +1,32 @@
+--------------- MODULE TestHash2 --------------
+\* Run Apalache to check the invariant for all combinations of hashes
+\*
```
Comment (konnov) [inline test/tla/TestHash2.tla:3]: `
\* This is an interesting example that appeared in the TLA+ user group:
\*(https://groups.google.com/g/tlaplus/c/cD1bhMcHxqA).
`

**Pull request merged**